---
title: "data exploration PE462"
author: "Emna Zeghal"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: yes
    toc_float: true
    toc_depth: 6
    code_folding: show
#  pdf_document:
 #   toc: yes
#  word_document:
    #toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Purpose
the project referred to as NIOZ186 is basically the exploration of Illumina sequencing data (primers 515F-926R). The amplified DNA was extracted from the surface of plastic films (polymers: PE, PET, PS and Nylon) incubated in 5L bottles of seawater collected from 2 different station (offshore vs coastal) in the North Sea.  
The files used here are the resulting asv table and taxonomy table obtained via CASCABEL (v4.6.2) (Abdala Asbun et al., 2020).    
Raw sequencing data, original asv table and taxonomy table are accessible via [link].  


## Pre-processing  
First, I started by creating a R project and initializing a renv lock. After downloading the several needed packages (from CRAN or bioconductor). Those need to be loaded.  
```{r load libraries, results='hide', message=FALSE}
library(phyloseq)
library(grid)
library(tidyverse)
library(vegan)
library(scales)
library(rmarkdown)
library(knitr)
library(DESeq2)
library(microbiome)
library(ggpubr)
library(FactoMineR)
library(factoextra)
library(usedist)
library(heatmaply)
library(Hmisc)
library(kableExtra)
library(RColorBrewer)
library(vegan)
library(pals)
```  

```{r set seed}
#For random iterations
set.seed(42)
```
### Importing data:  
Raw data is imported in the right class to merge as a phyloseq object.  
```{r Data import}
tax1 <- as.matrix(read.delim("./original/representative_seq_set_tax_assignments.txt", row.names = 1, na.strings = "NA"))
tax2 <- read.delim("./original/representative_seq_set_tax_assignments_unpaired.txt", row.names = 1, na.strings = "NA")
rownames(tax2) <- paste0("asv.",(length(rownames(tax1))+1):(length(rownames(tax1))+length(rownames(tax2))))
tax2 ['Species']= NA
tax2 <- as.matrix(tax2)
tax <- rbind(tax1,tax2) %>% tax_table()
otu1 <- as.matrix(read.delim("./original/asv_table.txt", row.names = 1))
otu2 <- read.delim("./original/asv_table_unpaired.txt", row.names = 1)
rownames(otu2) <- rownames(tax2)
samples_missing <- colnames(otu1)[c(which(colnames(otu1) %nin% colnames(otu2)))]
for (i in 1:length(samples_missing)) {otu2[samples_missing[i]] <- 0 }
otu2 <- as.matrix (otu2)
otu <-rbind(otu1,otu2) %>%  otu_table(taxa_are_rows = T)
map <- sample_data(read.delim("./original/mapPE462.txt", row.names = 1, na.strings = c("NA", "")))
physeq_object = merge_phyloseq(otu, tax, map)                 
```  

```{r physeq_object, collapse=TRUE}
physeq_object
```  

```{r basic information,collapse=TRUE}
##############basic information##############
summarize_phyloseq(physeq_object)
ntaxa(physeq_object)
nsamples(physeq_object) 
head(sample_names(physeq_object))
head(taxa_names(physeq_object))
rank_names(physeq_object)
head(sample_sums(physeq_object))
head(taxa_sums(physeq_object))
min(sample_sums(physeq_object))
#eliminate singletons
physeq_object <- filter_taxa(physeq_object, function(x) sum(x) > 1, TRUE) 
```  

```{r getting rid of wonky taxonomy assignments, collapse=TRUE }
##########getting rid of wonky taxonomy assignments ###########
get_taxa_unique(physeq_object, "Kingdom") 
physeq_object <- subset_taxa(physeq_object, !is.na(Kingdom) & !Kingdom%in% c("", "Unassigned"))
get_taxa_unique(physeq_object, "Kingdom")
###rid of NA in taxonomy
taxo <- as.data.frame(physeq_object@tax_table)

for (i in 1:nrow(taxo)) {
  for (y in 1:ncol(taxo)) {
    if 
    (is.na(taxo[i,y]) || any(str_detect(taxo[i,y], c("uncultured","Uncultured","metagenome", "Metagenome","unknown", "Unknown","NA")))) {
      taxo[i,y] <- "unassigned" }
  }
}

taxo <- tax_table(as.matrix(taxo))
physeq_object <- merge_phyloseq(physeq_object@otu_table, taxo, map)

###rid of chloroplast & Mitochodria
any((get_taxa_unique(physeq_object, "Order") == "Chloroplast"))
any((get_taxa_unique(physeq_object, "Family") == "Mitochondria"))
physeq_object <- subset_taxa(physeq_object, !Order%in% c("Chloroplast")) 
physeq_object <- subset_taxa(physeq_object, !Family%in% c("Mitochondria"))
any((get_taxa_unique(physeq_object, "Order") == "Chloroplast"))
any((get_taxa_unique(physeq_object, "Family") == "Mitochondria"))
#no singletons
physeq_object <- prune_taxa(taxa_sums(physeq_object) > 1, physeq_object) 
``` 

```{r Final phyloseq object}
physeq_object <- merge_phyloseq(physeq_object@otu_table, taxo, map)
```
```{r plot prevalence phyla, message=FALSE, echo=FALSE, fig.pos="center", size= 10, fig.align = 'center', fig.cap = "Prevalence of all detected Phyla"}
microbiome::plot_taxa_prevalence(physeq_object, "Phylum")+ theme(legend.position = "none")
```
### Alpha diversity:  
Let's calculate Alpha diversity indexes. I am using here two different packages. One is enough. I was just curious if the output would be the same.  

```{r Alpha_div, results='hide', message=FALSE}
############### alpha div ###################
#microbiome package alpha div indexes
summarize_phyloseq(physeq_object)
alpha_tab <-microbiome::alpha(physeq_object, index = "all")
alpha_tab <- alpha_tab %>% cbind(data.frame(physeq_object@sam_data))
#write_csv(alpha_tab, file = "./alpha_div_indexes_microbiome_package.csv")
#phyloseq package
alpha <- phyloseq::estimate_richness(physeq_object)
#write_csv(alpha, file = "./alpha_div_phyloseq.csv")
```

```{r alpha div tab, echo=FALSE}
knitr::kable(alpha_tab,"html", caption = "Alpha diversity idexes table", digits = 2) %>% kableExtra::scroll_box(width = "8tt00px", height = "300px") 
```  

Here's a few graphical representations for all samples. See 4_alpha_div_plots.R script.
```{r ,fig.fullwidth=TRUE, fig.align ='center', echo=FALSE, results='hide', message=FALSE}
alpha_tab <- read_csv("./new/alpha_div_indexes_microbiome_package.csv")
alpha_tab <- alpha_tab %>%  filter(station %in% c("C05", "C13")) %>% filter(material %nin% c("Glass_fiber")) %>% 
  filter(timepoint.days. %nin% c("15"))
alpha_tab$polymer <- if_else(alpha_tab$material=="wood", str_c(alpha_tab$material), alpha_tab$polymer)
alpha_tab_coast <- alpha_tab %>% filter(station %in% c("C13")) %>% mutate(across(c(treatment),factor)) %>% 
  mutate(across(c(treatment),factor))
alpha_tab_os <- alpha_tab %>% filter(station %in% c("C05")) %>% mutate(across(c(treatment),factor)) %>% 
  mutate(across(c(treatment),factor))
```

```{r ,fig.fullwidth=TRUE, fig.align ='center', message=FALSE, echo=FALSE}
###create graphs##
p_coast <-ggplot(alpha_tab_coast,aes(x = reorder(timepoint.days., as.numeric(timepoint.days.)), y = evenness_simpson, fill=treatment))+
  geom_boxplot(position=position_dodge(1))  +scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9"))+ theme_minimal()+
  facet_wrap(~polymer, nrow = 5)+
  labs(title="Alpha diversity Coastal station",x="incubation time (days)", y = "Simpson eveness index")+ ylim(0,0.35)
p_os <-ggplot(alpha_tab_os,aes(x = reorder(timepoint.days., as.numeric(timepoint.days.)), y = evenness_simpson, fill=treatment))+
  geom_boxplot(position=position_dodge(1)) +scale_y_continuous(limits = c(0,0.35))+ scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9"))+ theme_minimal()+
  facet_wrap(~polymer, nrow = 5)+
  labs(title="Alpha diversity open water station",x="incubation time (days)", y = "Simpson eveness index")
fig1 <- ggarrange(p_coast,p_os, common.legend = T) 
#ggexport(fig,filename = "simpson_index.pdf")
fig1

```  

```{r ,fig.fullwidth=TRUE, fig.align ='center'}
sh_coast <-ggplot(alpha_tab_coast,aes(x = reorder(timepoint.days., as.numeric(timepoint.days.)), y = diversity_shannon, fill=treatment))+
  geom_boxplot(position=position_dodge(1))+scale_y_continuous(limits = c(0,6))  +scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9"))+ theme_minimal()+
  facet_wrap(~polymer, nrow = 5)+
  labs(title="Coastal station",x="incubation time (days)", y = "Shannon diversity index")

sh_os <-ggplot(alpha_tab_os,aes(x = reorder(timepoint.days., as.numeric(timepoint.days.)), y = diversity_shannon, fill=treatment))+
  geom_boxplot(position=position_dodge(1))+scale_y_continuous(limits = c(0,6))  + scale_fill_manual(values=c("#999999", "#E69F00", "#56B4E9"))+ theme_minimal()+
  facet_wrap(~polymer, nrow = 5)+
  labs(title="Open water station",x="incubation time (days)", y = "Shannon diversity index")
fig2 <- ggarrange(sh_coast,sh_os, common.legend = T) 
#ggexport(fig,filename = "shannon_index.pdf")
fig2
```  

### A focus on the wooden stirrers controls:  
Let's try to determine if the Alpha diversity between the different bottles of incubation for the wooden stirrers are really different (for each timepoint and station of course). Combining those to "create" replicates could be interesting.  

#### Alpha diversity graphic glimpse:  

```{r wooden stirrers tab with alpha div, results='hide', message=FALSE}
alpha_tab <- read_csv("./new/alpha_div_indexes_microbiome_package.csv")
m <- alpha_tab %>% filter(material =="wood")
```
```{r boxplot alpha di wooden stirrers, fig.fullwidth=TRUE, fig.align ='center', echo=}
p <- ggboxplot(m, x = "timepoint.days.", y = "evenness_simpson",
               fill = "timepoint.days.", palette ="Dark2",
              add = "jitter", size = 1) +facet_wrap(~station)
p <- ggpar(p,legend = "none",xlab ="incubation time (days)", title = "Simpson eveness index")

p2 <- ggboxplot(m, x = "timepoint.days.", y = "diversity_shannon",
               fill = "timepoint.days.", palette ="Dark2",
              add = "jitter", size = 1) +facet_wrap(~station)
p2 <- ggpar(p2,legend = "none",xlab ="incubation time (days)", title = "Shannon diversity index")
ggarrange (p, p2, ncol=2)

```  

#### Kruskal wallis tests for each station/timepoint combination:  

##### coastal station / day 5:  
```{r kruskal C05/day5, collapse=TRUE}
kruskal.test(diversity_shannon ~ polymer, data = (m %>% filter (station =="C05") %>% filter(timepoint.days. == "5")))  
kruskal.test(evenness_simpson ~ polymer, data = (m %>% filter (station =="C05") %>% filter(timepoint.days. == "5")))
```
##### coastal station / day 10:  
```{r kruskal C05/day10, collapse=TRUE}
kruskal.test(diversity_shannon ~ polymer, data = (m %>% filter (station =="C05") %>% filter(timepoint.days. == "10")))  
kruskal.test(evenness_simpson ~ polymer, data = (m %>% filter (station =="C05") %>% filter(timepoint.days. == "10")))
```  

##### coastal station / day 45:  
```{r kruskal C05/day45, collapse=TRUE}
kruskal.test(diversity_shannon ~ polymer, data = (m %>% filter (station =="C05") %>% filter(timepoint.days. == "45")))  
kruskal.test(evenness_simpson ~ polymer, data = (m %>% filter (station =="C05") %>% filter(timepoint.days. == "45")))
```  
None of the tests above showed significant differences (p<0.05). Combining wooden stirrers for each timepoint might be considered.  

##### offshore station / day 5:  
```{r kruskal C13/day5, collapse=TRUE}
kruskal.test(diversity_shannon ~ polymer, data = (m %>% filter (station =="C13") %>% filter(timepoint.days. == "5")))  
kruskal.test(evenness_simpson ~ polymer, data = (m %>% filter (station =="C13") %>% filter(timepoint.days. == "5")))
```
##### offshore station / day 10:  
```{r kruskal C13/day10, collapse=TRUE}
kruskal.test(diversity_shannon ~ polymer, data = (m %>% filter (station =="C13") %>% filter(timepoint.days. == "10")))  
kruskal.test(evenness_simpson ~ polymer, data = (m %>% filter (station =="C13") %>% filter(timepoint.days. == "10")))
```  
##### offshore station / day 30:  
```{r kruskal C13/day30, collapse=TRUE}
kruskal.test(diversity_shannon ~ polymer, data = (m %>% filter (station =="C13") %>% filter(timepoint.days. == "30")))  
kruskal.test(evenness_simpson ~ polymer, data = (m %>% filter (station =="C13") %>% filter(timepoint.days. == "30")))
```  
##### offshore station / day 45:  
```{r kruskal C13/day45, collapse=TRUE}
kruskal.test(diversity_shannon ~ polymer, data = (m %>% filter (station =="C13") %>% filter(timepoint.days. == "45")))  
kruskal.test(evenness_simpson ~ polymer, data = (m %>% filter (station =="C13") %>% filter(timepoint.days. == "45")))
```  
None of the tests above showed significant differences (p<0.05). Combining wooden stirrers for each timepoint might be considered.  
According to the results above. wooden stirrers might be pooled for each station/timepoint combination.  

## Data transformation  
Now, I am transforming my data in a "tidy" way to make use of the function implemented in the tidyverse. I am mainly referring to grouping functions that will allow quick calculations of mean abundances and standard deviation for replicates without writing loops.  
For that I am going to use the tidy_psmelt function (written by mikemc github user). tidy_psmelt source code is saved in the project repository.  

### Tidying the data:
```{r tidy data, file="./tidy_psmelt.R", message=FALSE}
source("./tidy_psmelt.R")
tidy_physeq_asv <- tidy_psmelt(physeq_object)
tidy_physeq_asv <- tidy_physeq_asv %>% filter(!Phylum == "unassigned") %>% filter(timepoint.days. %nin% c("15")) # rid of unassigned phyla and timepoint 15 incomplete data set
tidy_physeq_asv$Species <- if_else(!tidy_physeq_asv$Species=="unassigned", str_c(tidy_physeq_asv$Genus," ",tidy_physeq_asv$Species), tidy_physeq_asv$Species)#append genus to the species name
tidy_physeq_asv$detail <- if_else(tidy_physeq_asv$material=="wood", str_c(tidy_physeq_asv$material,"_",tidy_physeq_asv$station,"_",tidy_physeq_asv$timepoint.days.), tidy_physeq_asv$detail)#correct column detail for wood to be able to merge the "replicates"
tidy_physeq_asv$polymer <- if_else(tidy_physeq_asv$material=="wood", str_c(tidy_physeq_asv$material), tidy_physeq_asv$polymer)#correct column detail for wood to be able to merge the "replicates"

#write_csv(tidy_physeq_asv, "./new/tidyPE462****.csv")
```  
### Abundances calculations:  
I am going to calculate the mean abundances of each agglomerated taxonomic rank per sample and per replicates.  
```{r abundances calc}
## Read last file created after "melting" phyloseq object
tidy_physeq_asv <- read.csv("./new/tidyPE462_v5.csv")

t3 <- tidy_physeq_asv  %>% group_by(Sample) %>% mutate(Sample_rel_abund = Abundance / sum(Abundance)) %>% #relative abundance of each asv per sample
  ungroup() %>%
  group_by(detail) %>% mutate( rep_rel_abund = Sample_rel_abund / sum(Sample_rel_abund)) %>% #relative abundance of each asv per number of samples in replicates
  ungroup() %>% 
  #Kingdom_section
  group_by(Sample, Kingdom) %>% 
  mutate(Kingdom_rel_abund_Sample = sum(Sample_rel_abund)) %>%  #Kingdom relative abundance per sample 
  ungroup() %>% 
  group_by(detail, Kingdom) %>% 
  mutate(Kingdom_st_dev_abund_samples = sd(Kingdom_rel_abund_Sample)) %>% # standard dev of Kingdom relative abundances between replicates of column detail 
  mutate(Kingdom_rep_rel_abund = sum(rep_rel_abund)) %>% #Kingdom relative abundance per samples of desc 
  ungroup() %>%
  #Phylum_section
  group_by(Sample, Phylum) %>% 
  mutate(Phylum_rel_abund_Sample = sum(Sample_rel_abund)) %>%
  ungroup() %>% 
  group_by(detail, Phylum) %>% 
  mutate(st_dev_Phylum_abund = sd(Phylum_rel_abund_Sample)) %>%
  mutate(Phyla_rep_rel_abund = sum(rep_rel_abund)) %>%
  ungroup() %>% 
  #Class_section
  group_by(Sample, Class) %>% 
  mutate(Class_rel_abund_Sample = sum(Sample_rel_abund)) %>%
  ungroup() %>% 
  group_by(detail, Class) %>% 
  mutate(st_dev_Class_abund = sd(Class_rel_abund_Sample)) %>%
  mutate(Class_rep_rel_abund = sum(rep_rel_abund)) %>%
  ungroup() %>% 
  #Order_section
  group_by(Sample, Order) %>% 
  mutate(Order_rel_abund_Sample = sum(Sample_rel_abund)) %>%
  ungroup() %>% 
  group_by(detail, Order) %>% 
  mutate(st_dev_Order_abund = sd(Order_rel_abund_Sample)) %>%
  mutate(Order_rep_rel_abund = sum(rep_rel_abund)) %>%
  ungroup() %>% 
  #Family_section
  group_by(Sample, Family) %>% 
  mutate(Family_rel_abund_Sample = sum(Sample_rel_abund)) %>%
  ungroup() %>% 
  group_by(detail, Family) %>% 
  mutate(st_dev_Family_abund = sd(Family_rel_abund_Sample)) %>%
  mutate(Family_rep_rel_abund = sum(rep_rel_abund)) %>%
  ungroup() %>% 
  #Genus_section
  group_by(Sample, Genus) %>% 
  mutate(Genus_rel_abund_Sample = sum(Sample_rel_abund)) %>%
  ungroup() %>% 
  group_by(detail, Genus) %>% 
  mutate(st_dev_Genus_abund = sd(Genus_rel_abund_Sample)) %>%
  mutate(Genus_rep_rel_abund = sum(rep_rel_abund)) %>%
  ungroup() %>% 
  #Species_section
  group_by(Sample, Species) %>% 
  mutate(Species_rel_abund_Sample = sum(Sample_rel_abund)) %>%
  ungroup() %>% 
  group_by(detail, Species) %>% 
  mutate(st_dev_Species_abund = sd(Species_rel_abund_Sample)) %>%
  mutate(Species_rep_rel_abund = sum(rep_rel_abund)) %>%
  ungroup()  

####final complete tidy Table with extra variables combination#####

polymer_station <-  str_c(t3$polymer, "_", t3$station)
polymer_photo <- str_c(t3$polymer, "_", t3$treatment)
pol_photo_station <- str_c(t3$polymer, "_", t3$treatment, "_", t3$station)

t3 <- t3 %>% 
  add_column(polymer_station, .before ="Kingdom") %>% 
  add_column(polymer_photo, .before ="Kingdom") %>% 
  add_column(pol_photo_station, .before ="Kingdom")

t3$polymer_station <- if_else(t3$material=="wood", str_c(t3$material,"_",t3$station), t3$polymer_station)
t3$polymer_photo <- if_else(t3$material=="wood", str_c(t3$material), t3$polymer_photo)
t3$pol_photo_station <- if_else(t3$material=="wood", str_c(t3$material,"_",t3$station), t3$pol_photo_station)

#write_csv(t3, "./new/tidyPE462_abundances_calc_v5.csv")  
 
```

```{r tidy data table, echo=FALSE}
knitr::kable(t3[1:10,],caption = "First 10 rows of the tidy transformed datset", digits = 2)%>%  kableExtra::scroll_box(width = "800px", height = "300px") 
```



## Community analysis  
I imported the resulting datasets after pre-processing for each polymer in the working project directory.  
A common first step is to load the libraries to be used throughout the analysis.

### Polyethylene:  

#### 1. Importing data:    
The datset is imported then split into several subsets according to different parameters (UV treatment and location)   
```{r import and split PE, results='hide', message=FALSE}
#PE_analysis==============
pe <- read_csv("./new/PE462_PE_v5.csv")
pe <- pe %>% filter(Genus %nin% c("unassigned", "Unassigned"))
    ######subset per station#########
coast <- pe %>% filter(station %in% c("C05"))
os <- pe %>% filter(station %in% c("C13"))
    #####subset stations per treatment######
coast_uv <- coast %>% filter(treatment %in% c("UV"))
coast_nuv <- coast %>% filter(treatment %in% c("no_UV"))
coast_n <- coast %>% filter(treatment %in% c("no"))
##same for 2nd station
os_uv <- os %>% filter(treatment %in% c("UV"))
os_nuv <- os %>% filter(treatment %in% c("no_UV"))
os_n <- os %>% filter(treatment %in% c("no"))
```
#### 2. Determining top genera:  
here I determined the top 5 genera for each subset of the coastal station and each timepoint.  
```{r top genera PE, results='hide', message=FALSE}
coast_uv_g5 <- coast_uv %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
coast_nuv_g5 <- coast_nuv %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
coast_n_g5 <- coast_n %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
```
These are the "total" top genera:  
```{r, collapse=TRUE} 
unique(c(coast_nuv_g5$Genus,coast_uv_g5$Genus, coast_n_g5$Genus) )
```
Now back to station subsets to extract the top genera for each  
```{r}
coast_top5 <- coast %>% select(timepoint.days.,detail,polymer_photo, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor)) %>% 
  distinct()%>% filter(Genus %in% (unique(c(coast_nuv_g5$Genus,coast_uv_g5$Genus, coast_n_g5$Genus))))
```

```{r heatmap pe coast, results='hide', message=FALSE}
no_white <- mf_get_pal(n = c( 1,30), pal = c("Light Grays","Burg" ), rev =c(F,F))

pe_coast_top5<- ggplot(data = coast_top5, mapping = aes(y = fct_relevel(Genus,rev), x = timepoint.days., fill = Genus_rep_rel_abund)) + 
  geom_tile()  +scale_fill_gradientn(name = "relative abundance", colours = no_white, limits=c(0,0.7))+
  facet_grid(~ polymer_photo, scales = "free_x", space = "free_x") + xlab(label = "incubation time (days)") +
  theme_bw() +
  theme(strip.placement = "outside",
        plot.title = element_text(hjust = 0.5),
        axis.title.y = element_blank(),
        axis.text.y = element_text(face = "italic"),
        strip.background = element_rect(fill = "#FFFFFF", color = "#FFFFFF")) +
  ggtitle(label = "Top genera abundance for PE in coastal station")

```
```{r plot heatmap pe coast, message=FALSE, fig.align='center'}
pe_coast_top5
```  
  
  

Now same logic for offshore station.  
```{r, results='hide', message=FALSE}
os_uv_g5 <- os_uv %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
os_nuv_g5 <- os_nuv %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
os_n_g5 <- os_n %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)

```
```{r top5 os pe} 
unique(c(os_nuv_g5$Genus,os_uv_g5$Genus, os_n_g5$Genus) )
```
```{r, results='hide', message=FALSE}
os_top5 <- os %>% select(timepoint.days.,detail,polymer_photo, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor)) %>% 
  distinct()%>% filter(Genus %in% (unique(c(os_nuv_g5$Genus,os_uv_g5$Genus, os_n_g5$Genus))))
```
```{r pe_os_top5, echo=FALSE}
pe_os_top5<- ggplot(data = os_top5, mapping = aes(y = Genus, x = timepoint.days., fill = Genus_rep_rel_abund)) + 
  geom_tile()  +scale_fill_gradientn(name = "relative abundance",colours = c("#FDFEFE","#A1DAB4" ,"#41B6C4" ,"#225EA8"), limits=c(0,0.6))+
  facet_grid(~ polymer_photo, scales = "free_x", space = "free_x") + xlab(label = "incubation time (days)") +
  theme_bw() +
  theme(strip.placement = "outside",
        plot.title = element_text(hjust = 0.5),
        axis.title.y = element_blank(),
        strip.background = element_rect(fill = "#EEEEEE", color = "#FFFFFF")) +
  ggtitle(label = "Top genera abundance for PE in offshore station") 
```
```{r plot heatmap pe offshore, message=FALSE, fig.align='center'}
plot(pe_os_top5)
```


### Polyethylene Tetraphtalate:  

#### 1. Importing data:   
Same as previous polymer.      
```{r, results='hide', message=FALSE}
###############data import#########
pet <- read_csv("./new/PE462_PET_v5.csv")
pet <- pet %>% filter(Genus %nin% c("unassigned", "Unassigned"))
############subset per station##########
coast <- pet %>% filter(station %in% c("C05"))
os <- pet %>% filter(station %in% c("C13"))
#####subset stations per treatment#####
coast_uv <- coast %>% filter(treatment %in% c("UV"))
coast_nuv <- coast %>% filter(treatment %in% c("no_UV"))
coast_n <- coast %>% filter(treatment %in% c("no"))
##same for 2nd station
os_uv <- os %>% filter(treatment %in% c("UV"))
os_nuv <- os %>% filter(treatment %in% c("no_UV"))
os_n <- os %>% filter(treatment %in% c("no"))
```
#### 2. Determining top genera:  
here I determined the top 5 genera for each subset of the coastal station and each timepoint.  
```{r, results='hide', message=FALSE}
coast_uv_g5 <- coast_uv %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
coast_nuv_g5 <- coast_nuv %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
coast_n_g5 <- coast_n %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
```
These are the "total" top genera:  
```{r} 
unique(c(coast_nuv_g5$Genus,coast_uv_g5$Genus, coast_n_g5$Genus) )
```
Now back to station subsets to extract the top genera for each  

```{r, results='hide', message=FALSE}
coast_top5 <- coast %>% select(timepoint.days.,detail,polymer_photo, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor)) %>% 
  distinct()%>% filter(Genus %in% (unique(c(coast_nuv_g5$Genus,coast_uv_g5$Genus, coast_n_g5$Genus))))
```
```{r pet_coast_top5, echo=FALSE}
pet_coast_top5<- ggplot(data = coast_top5, mapping = aes(y = Genus, x = timepoint.days., fill = Genus_rep_rel_abund)) + 
  geom_tile()  +scale_fill_gradientn(name = "relative abundance",colours = c("#FDFEFE","#A1DAB4" ,"#41B6C4" ,"#225EA8"), limits=c(0,0.6))+
  facet_grid(~ polymer_photo, scales = "free_x", space = "free_x") + xlab(label = "incubation time (days)") +
  theme_bw() +
  theme(strip.placement = "outside",
        plot.title = element_text(hjust = 0.5),
        axis.title.y = element_blank(),
        strip.background = element_rect(fill = "#EEEEEE", color = "#FFFFFF")) +
  ggtitle(label = "Top genera abundance for PET in coastal station") 
```
```{r plot heatmap pet coast, message=FALSE, fig.align='center'}
plot(pet_coast_top5)
```

```{r, results='hide', message=FALSE}
os_uv_g5 <- os_uv %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
os_nuv_g5 <- os_nuv %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
os_n_g5 <- os_n %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)

```
```{r} 
unique(c(os_nuv_g5$Genus,os_uv_g5$Genus, os_n_g5$Genus) )
```
```{r, results='hide', message=FALSE}
os_top5 <- os %>% select(timepoint.days.,detail,polymer_photo, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor)) %>% 
  distinct()%>% filter(Genus %in% (unique(c(os_nuv_g5$Genus,os_uv_g5$Genus, os_n_g5$Genus))))
```
```{r pet_os_top5, echo=FALSE}
pet_os_top5<- ggplot(data = os_top5, mapping = aes(y = Genus, x = timepoint.days., fill = Genus_rep_rel_abund)) + 
  geom_tile()  +scale_fill_gradientn(name = "relative abundance",colours = c("#FDFEFE","#A1DAB4" ,"#41B6C4" ,"#225EA8"), limits=c(0,0.6))+
  facet_grid(~ polymer_photo, scales = "free_x", space = "free_x") + xlab(label = "incubation time (days)") +
  theme_bw() +
  theme(strip.placement = "outside",
        plot.title = element_text(hjust = 0.5),
        axis.title.y = element_blank(),
        strip.background = element_rect(fill = "#EEEEEE", color = "#FFFFFF")) +
  ggtitle(label = "Top genera abundance for PET in coastal station") 
```
```{r plot heatmap pet offshore, message=FALSE, fig.align='center'}
plot(pet_os_top5)
```


### Polystyrene:  

#### 1. Importing data:    
Same as previous polymer.      
```{r, results='hide', message=FALSE}
###############data import#########
ps <- read_csv("./new/PE462_PS_v5.csv")
ps <- ps %>% filter(Genus %nin% c("unassigned", "Unassigned"))
############subset per station##########
coast <- ps %>% filter(station %in% c("C05"))
os <- ps %>% filter(station %in% c("C13"))
#####subset stations per treatment#####
coast_uv <- coast %>% filter(treatment %in% c("UV"))
coast_nuv <- coast %>% filter(treatment %in% c("no_UV"))
coast_n <- coast %>% filter(treatment %in% c("no"))
##same for 2nd station
os_uv <- os %>% filter(treatment %in% c("UV"))
os_nuv <- os %>% filter(treatment %in% c("no_UV"))
os_n <- os %>% filter(treatment %in% c("no"))
```
#### 2. Determining top genera:  
here I determined the top 5 genera for each subset of the coastal station and each timepoint.  
```{r, results='hide', message=FALSE}
coast_uv_g5 <- coast_uv %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
coast_nuv_g5 <- coast_nuv %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
coast_n_g5 <- coast_n %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
```
These are the "total" top genera:  
```{r} 
unique(c(coast_nuv_g5$Genus,coast_uv_g5$Genus, coast_n_g5$Genus) )
```
Now back to station subsets to extract the top genera for each  
```{r, results='hide', message=FALSE}
coast_top5 <- coast %>% select(timepoint.days.,detail,polymer_photo, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor)) %>% 
  distinct()%>% filter(Genus %in% (unique(c(coast_nuv_g5$Genus,coast_uv_g5$Genus, coast_n_g5$Genus))))
```
```{r ps_coast_top5, echo=FALSE}
ps_coast_top5<- ggplot(data = coast_top5, mapping = aes(y = Genus, x = timepoint.days., fill = Genus_rep_rel_abund)) + 
  geom_tile()  +scale_fill_gradientn(name = "relative abundance",colours = c("#FDFEFE","#A1DAB4" ,"#41B6C4" ,"#225EA8"), limits=c(0,0.6))+
  facet_grid(~ polymer_photo, scales = "free_x", space = "free_x") + xlab(label = "incubation time (days)") +
  theme_bw() +
  theme(strip.placement = "outside",
        plot.title = element_text(hjust = 0.5),
        axis.title.y = element_blank(),
        strip.background = element_rect(fill = "#EEEEEE", color = "#FFFFFF")) +
  ggtitle(label = "Top genera abundance for PS in coastal station") 
```
```{r plot heatmap ps coast, message=FALSE, fig.align='center'}
plot(ps_coast_top5)
``` 

```{r, results='hide', message=FALSE}
os_uv_g5 <- os_uv %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
os_nuv_g5 <- os_nuv %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
os_n_g5 <- os_n %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)

```
```{r} 
unique(c(os_nuv_g5$Genus,os_uv_g5$Genus, os_n_g5$Genus) )
```
```{r, results='hide', message=FALSE}
os_top5 <- os %>% select(timepoint.days.,detail,polymer_photo, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor)) %>% 
  distinct()%>% filter(Genus %in% (unique(c(os_nuv_g5$Genus,os_uv_g5$Genus, os_n_g5$Genus))))
```
```{r ps_os_top5, echo=FALSE}
ps_os_top5<- ggplot(data = os_top5, mapping = aes(y = Genus, x = timepoint.days., fill = Genus_rep_rel_abund)) + 
  geom_tile()  +scale_fill_gradientn(name = "relative abundance",colours = c("#FDFEFE","#A1DAB4" ,"#41B6C4" ,"#225EA8"), limits=c(0,0.6))+
  facet_grid(~ polymer_photo, scales = "free_x", space = "free_x") + xlab(label = "incubation time (days)") +
  theme_bw() +
  theme(strip.placement = "outside",
        plot.title = element_text(hjust = 0.5),
        axis.title.y = element_blank(),
        strip.background = element_rect(fill = "#EEEEEE", color = "#FFFFFF")) +
  ggtitle(label = "Top genera abundance for PS in offshore station") 
```
```{r plot heatmap ps offshore, message=FALSE, fig.align='center'}
plot(ps_os_top5)
```

### Nylon:  

#### 1. Importing data:    
Same as previous polymer.      
```{r, results='hide', message=FALSE}
###############data import#########
Nylon <- read_csv("./new/PE462_Nylon_v5.csv")
Nylon <- Nylon %>% filter(Genus %nin% c("unassigned", "Unassigned"))
############subset per station##########
coast <- Nylon %>% filter(station %in% c("C05"))
os <- Nylon %>% filter(station %in% c("C13"))
#####subset stations per treatment#####
coast_uv <- coast %>% filter(treatment %in% c("UV"))
coast_nuv <- coast %>% filter(treatment %in% c("no_UV"))
coast_n <- coast %>% filter(treatment %in% c("no"))
##same for 2nd station
os_uv <- os %>% filter(treatment %in% c("UV"))
os_nuv <- os %>% filter(treatment %in% c("no_UV"))
os_n <- os %>% filter(treatment %in% c("no"))
```
#### 2. Determining top genera:  
here I determined the top 5 genera for each subset of the coastal station and each timepoint.  
```{r, results='hide', message=FALSE}
coast_uv_g5 <- coast_uv %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
coast_nuv_g5 <- coast_nuv %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
coast_n_g5 <- coast_n %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
```
These are the "total" top genera:  
```{r} 
unique(c(coast_nuv_g5$Genus,coast_uv_g5$Genus, coast_n_g5$Genus) )
```
Now back to station subsets to extract the top genera for each  

```{r, results='hide', message=FALSE}
coast_top5 <- coast %>% select(timepoint.days.,detail,polymer_photo, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor)) %>% 
  distinct()%>% filter(Genus %in% (unique(c(coast_nuv_g5$Genus,coast_uv_g5$Genus, coast_n_g5$Genus))))
```
```{r Nylon_coast_top5,echo=FALSE}
Nylon_coast_top5<- ggplot(data = coast_top5, mapping = aes(y = Genus, x = timepoint.days., fill = Genus_rep_rel_abund)) + 
  geom_tile()  +scale_fill_gradientn(name = "relative abundance",colours = c("#FDFEFE","#A1DAB4" ,"#41B6C4" ,"#225EA8"), limits=c(0,0.6))+
  facet_grid(~ polymer_photo, scales = "free_x", space = "free_x") + xlab(label = "incubation time (days)") +
  theme_bw() +
  theme(strip.placement = "outside",
        plot.title = element_text(hjust = 0.5),
        axis.title.y = element_blank(),
        strip.background = element_rect(fill = "#EEEEEE", color = "#FFFFFF")) +
  ggtitle(label = "Top genera abundance for Nylon in coastal station") 
```
```{r plot heatmap nylon coast, message=FALSE, fig.align='center'}
plot(Nylon_coast_top5)
```

```{r, results='hide', message=FALSE}
os_uv_g5 <- os_uv %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
os_nuv_g5 <- os_nuv %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)
os_n_g5 <- os_n %>% select(timepoint.days.,detail, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate(across(c(timepoint.days.),factor))%>% distinct() %>% 
  group_by(timepoint.days.) %>% slice_max(order_by = Genus_rep_rel_abund, n = 5)

```
```{r} 
unique(c(os_nuv_g5$Genus,os_uv_g5$Genus, os_n_g5$Genus) )
```
```{r, results='hide', message=FALSE}
os_top5 <- os %>% select(timepoint.days.,detail,polymer_photo, Genus, Genus_rep_rel_abund, st_dev_Genus_abund) %>% 
  mutate( polymer_photo = replace(polymer_photo, polymer_photo == "Nylon_wood_no", "ctrl_wood")) %>% mutate(across(c(timepoint.days.),factor)) %>% 
  distinct()%>% filter(Genus %in% (unique(c(os_nuv_g5$Genus,os_uv_g5$Genus, os_n_g5$Genus))))
```
```{r Nylon_os_top5, echo=FALSE}
Nylon_os_top5<- ggplot(data = os_top5, mapping = aes(y = Genus, x = timepoint.days., fill = Genus_rep_rel_abund)) + 
geom_tile()  +scale_fill_gradientn(name = "relative abundance",colours = c("#FDFEFE","#A1DAB4" ,"#41B6C4" ,"#225EA8"), limits=c(0,0.6))+
  facet_grid(~ polymer_photo, scales = "free_x", space = "free_x") + xlab(label = "incubation time (days)") +
  theme_bw() +
  theme(strip.placement = "outside",
        plot.title = element_text(hjust = 0.5),
        axis.title.y = element_blank(),
        strip.background = element_rect(fill = "#EEEEEE", color = "#FFFFFF")) +
  ggtitle(label = "Top genera abundance for Nylon in offshore station") 
```
```{r plot heatmap nylon offshore, message=FALSE, fig.align='center'}
plot(Nylon_os_top5)
```  

### Beta Diversity:  
I performed an nmds on the Genus dataset.
```{r physeq gen, results='hide'}
gen <- read_csv("./new/genus_PE462_v5.csv")
gen_count <- gen %>% filter(detail %nin% c("mock_DNA","Glass_fiber_no_C05_0", "Neg_PCR", "Glass_fiber_no_C13_0" )) %>% 
  select(detail, Genus, Genus_rep_rel_abund) %>% mutate(across(c(Genus),factor)) %>%
  pivot_wider(names_from = detail, values_from = Genus_rep_rel_abund, values_fill = 0) %>% column_to_rownames(var = "Genus") %>% as.matrix()
gen_metadata <- gen %>% select( "detail","station","timepoint.days.","treatment", "polymer","pol_photo_station","polymer_station",
                                "polymer_photo","material") %>% filter(detail %nin% c("mock_DNA","Glass_fiber_no_C05_0", "Neg_PCR", 
                                "Glass_fiber_no_C13_0" )) %>% distinct() %>% column_to_rownames(var = "detail")
pseq <- merge_phyloseq(otu_table(gen_count, taxa_are_rows = T), sample_data(gen_metadata))
```
```{r gen ord }
ord <- ordinate(pseq, method = "NMDS", k = 2, "bray",trymax = 150)
```
```{r stressplot, fig.align='center'}
stressplot(ord)
```  
The Stressplot above indicates that two dimensions aren't enough to represent data variation.  
Below, ord$points shows the coordinates of the points on the two axes MDS1 and MDS2.  
```{r gen ord coordinates }
head(ord$points)
```
```{r plot nmds 1,  message=FALSE, fig.align='center', echo=FALSE}
p <- plot_ordination(pseq, ord, color="polymer", shape="station") + geom_point(size=4, alpha=0.75)+ 
  scale_colour_brewer(type="qual", palette="Dark2") + ggtitle("NMDS on dissimilarity matrix for dataset agglomerated on genus level")
p
```


```{r plot nmds 2,  message=FALSE, fig.align='center', echo=FALSE}

p2 <- plot_ordination(pseq, ord, color="polymer_station", shape="timepoint.days.") + geom_point(size=4, alpha=0.75)+ 
  scale_colour_brewer(type="qual", palette="Paired") + ggtitle("NMDS on dissimilarity matrix for dataset agglomerated on genus level")
p2
```



## Conclusion:  
```{r stop }

#knitr::knit_exit()
``` 